### ISSUE TITLE
[Bug]: vllm==0.8.1, when I use : v1 + lora + allow_tokens_ids_mask,  Sampler.forward.apply_allowed_token_ids. func,  sampling_metadata.allowed_token_ids_mask.shape not match logits.shape

### ISSUE THREAD

**Issue by binbreaze** [vllm-project/vllm#15978]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Your output of `python collect_env.py` here
```
when I use : v1 + lora + allow_tokens_ids_mask,  sampling_metadata.allowed_token_ids_mask.shape not match logits.shape;

   def apply_allowed_token_ids(
        self,
        logits: torch.Tensor,
        sampling_metadata: SamplingMetadata,
    ) -> torch.Tensor:
        if sampling_metadata.allowed_token_ids_mask is not None:
            logits.masked_fill_(sampling_metadata.allowed_token_ids_mask,
                                float("-inf"))
        return logits

       -------------
       logits.shape = [num_req, vocab_size+lora_config.lora_extra_vocab_size],  then allowed_token_ids_mask.shape = [num_req, vocab_size]

</details>

### 🐛 Describe the bug

```
## vllm/v1/sample/sampler.py 

   def apply_allowed_token_ids(
        self,
        logits: torch.Tensor,
        sampling_metadata: SamplingMetadata,
    ) -> torch.Tensor:
        if sampling_metadata.allowed_token_ids_mask is not None:
            logits.masked_fill_(sampling_metadata.allowed_token_ids_mask,
                                float("-inf"))
        return logits


       -------------
       logits.shape = [num_req, vocab_size+lora_config.lora_extra_vocab_size],  then allowed_token_ids_mask.shape = [num_req, vocab_size]
```

```
## vllm/model_executor/models/llama.py 
class LlamaFotCausalLM:
      def __init__():
            if lora_config:
                self.unpadded_vocab_size += lora_config.lora_extra_vocab_size
 .........
logits.shape==self.unpadded_vocab_size
 but, 
 ##  vllm/v1/worker/gpu_input_batch.py 
 class InputBatch:
 def add_request():
         if sampling_params.allowed_token_ids:
            self.has_allowed_token_ids.add(req_id)
            if self.allowed_token_ids_mask_cpu_tensor is None:
                # Lazy allocation for this tensor, which can be large.
                # False means we don't fill with -inf.
                self.allowed_token_ids_mask = torch.zeros(self.max_num_reqs,
                                                          self.vocab_size,
                                                          dtype=torch.bool,
                                                          device=self.device)
                self.allowed_token_ids_mask_cpu_tensor = torch.zeros(
                    self.max_num_reqs,
                    self.vocab_size,
                    dtype=torch.bool,
                    device="cpu")
   .......
    allowed_token_ids_mask.shape == vocab_size
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by binbreaze** [comment]
## vllm/model_executor/models/llama.py 
class LlamaFotCausalLM:

      def __init__():
            if get_pp_group().is_last_rank:
                    self.unpadded_vocab_size = config.vocab_size
                    if lora_config:
                             self.unpadded_vocab_size += lora_config.lora_extra_vocab_size
 .........
logits.shape == self.unpadded_vocab_size ==  config.vocab_size+lora_config.lora_extra_vocab_size
 but, 
 ##  vllm/v1/worker/gpu_input_batch.py 
 class InputBatch:
 def add_request():
         if sampling_params.allowed_token_ids:
            self.has_allowed_token_ids.add(req_id)
            if self.allowed_token_ids_mask_cpu_tensor is None:
                # Lazy allocation for this tensor, which can be large.
                # False means we don't fill with -inf.
                self.allowed_token_ids_mask = torch.zeros(self.max_num_reqs,
                                                          self.vocab_size,
                                                          dtype=torch.bool,
                                                          device=self.device)
                self.allowed_token_ids_mask_cpu_tensor = torch.zeros(
                    self.max_num_reqs,
                    self.vocab_size,
                    dtype=torch.bool,
                    device="cpu")
   .......
    allowed_token_ids_mask.shape == vocab_size
